lib/Search/BasicSearch/BasicSearchEngine.php

```diff
@@ -0,0 +1,526 @@
+                        // creation
+                        $listViewDefs[$seed->module_dir][$listViewCheckField] = $orig_listViewDefs[$seed->module_dir][$listViewCheckField];
+                    }
+
+                    if (!empty($def['innerjoin'])) {
+                        if (empty($def['db_field'])) {
+                            continue;
+                        }
+                        $def['innerjoin'] = str_replace('INNER', 'LEFT', $def['innerjoin']);
+                    }
+
+                    if (isset($seed->field_defs[$field]['type'])) {
+                        $type = $seed->field_defs[$field]['type'];
+                        if ($type === 'int' && !is_numeric($searchQuery)) {
+                            continue;
+                        }
+                    }
+
+                    $unifiedSearchFields[$moduleName] [$field] = $def;
+                    $unifiedSearchFields[$moduleName] [$field]['value'] = $searchQuery;
+                }
+
+                /*
+                 * Use searchForm2->generateSearchWhere() to create the search query, as it can generate SQL for the full set of comparisons required
+                 * generateSearchWhere() expects to find the search conditions for a field in the 'value' parameter of the searchFields entry for that field
+                 */
+                require_once $beanFiles[$beanName];
+                $seed = new $beanName();
+
+                require_once $this->searchFormPath;
+                $searchForm = new $this->searchFormClass($seed, $moduleName);
+
+                $searchForm->setup(
+                    [$moduleName => []],
+                    $unifiedSearchFields,
+                    '',
+                    'saved_views'
+                );
+                $where_clauses = $searchForm->generateSearchWhere();
+                //add inner joins back into the where clause
+                $params = ['custom_select' => ""];
+                foreach ($innerJoins as $field => $def) {
+                    if (isset($def['db_field'])) {
+                        foreach ($def['db_field'] as $dbfield) {
+                            $where_clauses[] = $dbfield . " LIKE '" . DBManagerFactory::getInstance()->quote($searchQuery) . "%'";
+                        }
+                        $params['custom_select'] .= ", $dbfield";
+                        $params['distinct'] = true;
+                    }
+                }
+
+                if (!empty($where_clauses)) {
+                    $where = '((' . implode(' ) OR ( ', $where_clauses) . '))';
+                } else {
+                    $where = '';
+                }
+
+                $listData = $listViewData->getListViewData($seed, $where, 0, -1, [], $params);
+
+                $moduleCounts[$moduleName] = $listData['pageData']['offsets']['total'];
+
+                foreach ($listData['data'] as $i => $array) {
```

Comment:
Hi @Dillon-Brown if could also rename `$array` to `$hit` or `$entry` or `$result`, please?